index.js

```diff
@@ -2,24 +2,24 @@
 const os = require('os');
 const fs = require('fs');
 
+const MS_RE = /microsoft/i;
+
 const isWsl = () => {
 	if (process.platform !== 'linux') {
 		return false;
 	}
 
-	if (os.release().includes('Microsoft')) {
+	if (os.release().match(MS_RE) !== null) {
 		return true;
 	}
 
 	try {
-		return fs.readFileSync('/proc/version', 'utf8').includes('Microsoft');
+		if (fs.readFileSync('/proc/version', 'utf8').match(MS_RE) !== null) {
+			return true;
+		}
 	} catch (_) {
 		return false;
 	}
 };
 
-if (process.env.__IS_WSL_TEST__) {
-	module.exports = isWsl;
-} else {
-	module.exports = isWsl();
-}
+module.exports = isWsl;
```

Comment:
Exporting the function is a breaking change. I don't want that.

Reply:
Well, then there is no portable and stable way of testing with mocking `process.platform`. 
Your trickery:
```javascript
const origPlatform = process.platform;
Object.defineProperty(process, 'platform', {value: 'linux'});
```
is not leaking to imported module - it's not portable with Windows/WSL and last test always fails.

Reply:
What if you switch to use `os.platform()` instead, which works the same as `process.platform`.

Reply:
Should work with `proxyquire`.

Reply:
Forget, I'm just stupid. We need to guard last "falsy" test case with:
```javascript
process.env.__IS_WSL_TEST__ = true;
// TEST
delete process.env.__IS_WSL_TEST__;
```
and its just normally works in test module scope.